### DISCUSSION TITLE
How do you run the integration tests locally

### DISCUSSION THREAD

**Discussion by JoA-MoS** [semantic-release/semantic-release#2354] [Q&A]
I have never been able to run the integration tests locally and have always had to push to the CI server to get them to run. I run windows and execute the test via WSL2 terminal with docker installed.

The error I get is

```
✖ integration › before hook Rejected promise returned by test
✖ integration › after.always hook Rejected promise returned by test
```

with detailed errors of :
```
OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: rootfs_linux.go:76: mounting "/home/joamos/code/github/semantic-release-upstream/test/helpers/config.yaml" to rootfs at "/verdaccio/conf/config.yaml" caused: mount through procfd: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
```

```
❯ docker --version
Docker version 20.10.11, build dea9396

❯ docker images
REPOSITORY               TAG       IMAGE ID       CREATED        SIZE
verdaccio/verdaccio      4         256a88b43be8   7 months ago   178MB
jamesdbloom/mockserver   latest    466141786e6c   2 years ago    154MB
pvdlg/docker-gitbox      latest    39b2b41ccdcb   3 years ago    317MB
```

I have run `docker system prune -a` and `docker volume prune` but still have the same result.  

Any advice would be much appreciated.

**Comment by travi** [comment]
i'm not aware of an issue that would result in what you are seeing. running `npm test` on a machine that has docker available should be enough.

our core maintainers use macOS and our CI runs in ubuntu. i would expect wsl2 to behave similarly, but i suppose it is possible that there is something different there that could cause a problem? if you discover a solution that suggests a change being needed on our end, we'd be open to adjustments for better compatibility, but i'm afraid i don't have much insight into how to investigate your issue much further.

**Reply by JoA-MoS** [reply]
Thanks @travi, I will see about trying on a mac and see if that makes a difference. If so I may play around with how it is mounting/binding the config with dockerode .

**Reply by JoA-MoS** [reply]
I can confirm it does run successfully on a Mac, I will do some research and see what I can find.
